components/org.wso2.carbon.identity.conditional.auth.functions.http/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/http/HTTPGetFunctionImpl.java

```diff
@@ -39,10 +41,38 @@ public HTTPGetFunctionImpl() {
     }
 
     @Override
-    public void httpGet(String epUrl, Map<String, Object> eventHandlers) {
+    public void httpGet(String epUrl, Object... params) {
+
+        Map<String, Object> eventHandlers;
+        Map<String, String> headers = new HashMap<>();
+
+        switch (params.length) {
+            case 1:
+                if (params[0] instanceof Map) {
+                    eventHandlers = (Map<String, Object>) params[0];
+                } else {
+                    throw new IllegalArgumentException("Invalid argument type.");
+                }
+                break;
+            case 2:
+                if (params[0] instanceof Map && params[1] instanceof Map ) {
+                    headers = (Map<String, String>) params[0];
+                    eventHandlers = (Map<String, Object>) params[1];
+                } else {
+                    throw new IllegalArgumentException("Invalid argument type.");
```

Comment:
Let's elaborate more on the error messages and lets have different meaningful error messages per defined cases.

Reply:
Added in [f529f98](https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/140/commits/f529f9879dc8d5b7ac378837a5f1c06d1a1c4a2a).

The expression `entry.getValue().getClass().getName()` returns the class name 
jdk.nashorn.api.scripting.ScriptObjectMirror in the context of JavaScript Objects, which is not the expected result. Therefore, the error message has been refined to only mention the expected type:

`throw new IllegalArgumentException("Invalid argument types. Expected payloadData and eventHandlers " +
                            "(both of type Map<String, Object>) respectively.");`
                            
Further, I checked with the current implementation of other function and following is how the `executeStep` method logs the error. 

```
public void executeStep(int stepId, Object... params) {
       // blaa blaa
        if (params.length > 0) {
            // if there are any params provided, last one is assumed to be the event listeners
            if (params[params.length - 1] instanceof Map) {
                attachEventListeners((Map<String, Object>) params[params.length - 1]);
            } else {
                log.error("Invalid argument and hence ignored. Last argument should be a Map of event listeners.");
            }
        }
        if (params.length == 2) {
           
        }
    }
    ```